edit/api.go

```diff
@@ -281,3 +284,73 @@ func outputComplexCandidate(ec *eval.EvalCtx, a []eval.Value, o map[string]eval.
 	out := ec.OutputChan()
 	out <- c
 }
+
+func narrowRead(ec *eval.EvalCtx, a []eval.Value, o map[string]eval.Value) {
+	var (
+		source, action eval.Value
+		nr             = &narrow{}
+	)
+
+	eval.ScanArgs(a, &source, &action)
+	eval.TakeNoOpt(o)
+
+	ed := ec.Editor.(*Editor)
+
+	valueToItem := func(v eval.Value) narrowItem {
+		switch raw := v.(type) {
+		case eval.String:
+			return narrowStringItem(string(raw))
+		case eval.Map:
+		default:
+		}
+		return nil
+	}
+
+	switch rawSource := source.(type) {
+	case eval.CallableValue:
```

Comment:
The idea behind it is to support both static source and dynamic source. The latter allows itself to be called whenever the filter changes.